packages/ember-routing/lib/system/router.js

```diff
@@ -1109,20 +1109,21 @@ function forEachRouteAbove(originRoute, handlerInfos, callback) {
 // and are not meant to be overridable.
 let defaultActionHandlers = {
 
-  willResolveModel(transition, originRoute) {
-    originRoute.router._scheduleLoadingEvent(transition, originRoute);
+  willResolveModel(handlerInfos, transition, originRoute) {
+    this._scheduleLoadingEvent(transition, originRoute);
   },
 
   // Attempt to find an appropriate error route or substate to enter.
-  error(error, transition, originRoute) {
-    let handlerInfos = transition.state.handlerInfos;
-    let router = originRoute.router;
```

Comment:
Calling `originRoute.router` triggers the `cannot access 'router' of undefined` error I mentioned in one of the commit messages when a handler is loaded via async (or threw an error during its `init`).